crates/autopilot/src/run_loop.rs

```diff
@@ -432,28 +470,31 @@ impl RunLoop {
         // We do keep track of hashes we have already seen to reduce load from the node.
 
         let mut seen_transactions: HashSet<H256> = Default::default();
-        loop {
-            // This could be a while loop. It isn't, because some care must be taken to not
-            // accidentally keep the borrow alive, which would block senders. Technically
-            // this is fine with while conditions but this is clearer.
-            if self.current_block.borrow().number > deadline {
-                break;
-            }
+        while {
+            let block = self.current_block.borrow();
+            let still_time = block.number <= deadline;
+
+            // Take extra care to not accidentally keep the borrow alive within
+            // the `while` body, which would block senders.
```

Comment:
So, my understanding is that this is related to where the compiler places the `Drop`. For example, see [this issue](https://github.com/rust-lang/rust/issues/37612).

That being said, I _don't think_ it was a problem in our usage here, I just adapted the existing unbounded `loop` to be a `while` loop and just reworded the comment that was already there (I assumed it was done this way for a reason, I didn't verify it as such):

https://github.com/cowprotocol/services/blob/4dbeeb8170c52fe76b02f8730d30064f3426ad35/crates/autopilot/src/run_loop.rs#L435-L441

Reply:
I think we should review whether this is actually needed (I also think it's not in this case) just to avoid further confusion in the future.
I'm fine with leaving this as is since this PR is urgent.